restcomm/restcomm.ussd/src/main/java/org/restcomm/connect/ussd/telephony/UssdCallManager.java

```diff
@@ -228,45 +245,65 @@ private boolean redirectToHostedVoiceApp(final ActorRef self, final SipServletRe
+                    builder.setEmailAddress(account.getEmailAddress());
+                    final Sid sid = number.getUssdApplicationSid();
+                    if (sid != null) {
+                        final Application application = applications.getApplication(sid);
+                        RcmlserverConfigurationSet rcmlserverConfig = RestcommConfiguration.getInstance().getRcmlserver();
+                        RcmlserverResolver resolver = RcmlserverResolver.getInstance(rcmlserverConfig.getBaseUrl(), rcmlserverConfig.getApiPath());
+                        builder.setUrl(UriUtils.resolve(resolver.resolveRelative(application.getRcmlUrl())));
+                    } else {
+                        builder.setUrl(UriUtils.resolve(number.getUssdUrl()));
+                    }
+                    final String ussdMethod = number.getUssdMethod();
+                    if (ussdMethod == null || ussdMethod.isEmpty()) {
+                        builder.setMethod("POST");
+                    } else {
+                        builder.setMethod(ussdMethod);
+                    }
+                    if (number.getUssdFallbackUrl() != null)
+                        builder.setFallbackUrl(number.getUssdFallbackUrl());
+                    builder.setFallbackMethod(number.getUssdFallbackMethod());
+                    builder.setStatusCallback(number.getStatusCallback());
+                    builder.setStatusCallbackMethod(number.getStatusCallbackMethod());
+                    final Props props = UssdInterpreter.props(builder.build());
+                    final ActorRef ussdInterpreter = getContext().actorOf(props);
+                    final ActorRef ussdCall = ussdCall();
+                    ussdCall.tell(request, self);
+
+                    ussdInterpreter.tell(new StartInterpreter(ussdCall), self);
+
+                    SipApplicationSession applicationSession = request.getApplicationSession();
+                    applicationSession.setAttribute("UssdCall", "true");
+                    applicationSession.setAttribute(UssdInterpreter.class.getName(), ussdInterpreter);
+                    applicationSession.setAttribute(UssdCall.class.getName(), ussdCall);
+                    isFoundHostedApp = true;
                 } else {
-                    builder.setMethod(ussdMethod);
+                    if (logger.isDebugEnabled()) {
+                        final String errMsg = "Inbound USSD session is not Allowed";
+                        logger.debug(errMsg);
+                    }
+                    String errMsg = "Inbound USSD session to Number: " + number.getPhoneNumber()
+                            + " is not allowed";
+
+                    sendNotification(errMsg, 11001, "warning", true);
+                    final SipServletResponse resp = request.createResponse(SC_FORBIDDEN, "Inbound USSD session is not Allowed");
+                    resp.send();
+                    ec.executePostOutboundAction(far, extensions);
```

Comment:
are we invoking postAction here if feature isAllowed in the other branch?

Reply:
@jaimecasero we need to invoke `post` action in any case, I will fix that